test/dtypes/test_api_deprecation_warning.py

```diff
@@ -0,0 +1,85 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD 3-Clause license found in the
+# LICENSE file in the root directory of this source tree.
+
+"""
+Tests for deprecated API imports that have been moved to prototype.
+TODO: Remove these tests once the deprecated APIs have been removed.
+"""
+
+import sys
+import warnings
+
+
+def test_int8_dynamic_act_int4_weight_cpu_layout_deprecated():
+    """Test deprecation warning for Int8DynamicActInt4WeightCPULayout."""
+    # We need to clear the cache to force re-importing and trigger the warning again.
+    modules_to_clear = [
+        "torchao.dtypes.uintx.dyn_int8_act_int4_wei_cpu_layout",
+        "torchao.dtypes",
+    ]
+    for mod in modules_to_clear:
+        if mod in sys.modules:
+            del sys.modules[mod]
```

Comment:
maybe easier to just reset the warnings like this:
https://github.com/pytorch/ao/blob/6259e9885f0f0a9124dfbb7cd23bdcbf30bb1984/test/test_utils.py#L46